frontend/src/modules/GroupsModule.tsx

```diff
@@ -0,0 +1,411 @@
+                                className="px-1 py-2 hover:bg-gray-100 cursor-pointer"
+                                onClick={() => alert('Remove from view selected')}
+                            >
+                                Remove from view
+                            </li>
+                        </ul>
+                    </div>
+                )}
+            </div>
+
+            {/* Thumbnail Image */}
+            <img
+                src={imageUrl}
+                alt={`${groupName} Thumbnail`}
+                className="absolute top-2 left-2 w-12 h-12 object-cover z-10"
+            />
+
+            {/* Group Info */}
+            <div className="p-10 pl-6 z-0 flex-grow flex flex-col justify-end">
+                <h3 className="text-lg font-semibold text-primary-black">{groupName}</h3>
+            </div>
+        </div>
+    );
+}
+
+
+export default function GroupsModule() {
+    const userInfo = useContext<UserInfo | undefined>(UserInfoContext);
+    const currentDate = new Intl.DateTimeFormat('en-US', {
+        month: 'long',
+        day: 'numeric',
+        year: 'numeric'
+    }).format(new Date());
+
+    // If userInfo is not defined or the user is not a student or teacher, render nothing
+    if (!userInfo || (userInfo?.role !== "student" && userInfo?.role !== "teacher")) {
+        return <></>;
+    }
+
+
+    // const handleSubmit = (e: React.FormEvent) => {
+    //     e.preventDefault(); // Prevent default form submission
+    //     const confirmSubmit = window.confirm("Are you sure you want to submit your ratings?");
+
+    //     if (confirmSubmit) {
+    //         // Handle the actual submission logic here
+    //         console.log("Ratings submitted");
+    //         alert("Submission completed"); // Show an alert on successful submission
+    //     } else {
+    //         console.log("Submission cancelled");
+    //         alert("Submission cancelled"); // Show an alert if cancelled
+    //     }
+    // };
+
+
+    // Render for student
+    if (userInfo?.role === "student") {
+        return (
+            <div className="p-1 space-y-6">
+                {/* Welcome Section */}
+                <div className="bg-gradient-to-bl from-primary-red/60 to-primary-red rounded-lg shadow-md p-4 flex items-center">
+                    <div className="pl-8 flex-1">
+                        <p className="text-sm text-gray-200 mb-12">{currentDate}</p>
+                        
+                        <h1 className="text-3xl font-bold text-white">Welcome back, {userInfo.firstname}!</h1>
+                        <p className="text-sm text-gray-200 mt-1">Always stay updated in your student portal</p>
+                    </div>
+                    <img src="/src/assets/student.png" alt="Welcome" className="pr-7 w-200 h-40 object-contain" />
+                </div>
+
+                {/* My Classes Section */}
+                <div className="bg-primary-red/90 rounded-lg shadow-md p-6">
```

Comment:
Hey Sam, looking good so far but it seems like the problem currently is that Mouhamed already implemented the UI and logic behind the Course module as that was his task assigned to him. We do like your changes to the Welcome module for Students, so if possible please save line 151 to line 160 of this file, do `git pull origin main`, take all incoming changes, add that code and make a new commit to this PR.

If you're still confused, please contact me or @JayC-SF for assistance. Thanks! 😄

Reply:
yeah what he said